### ISSUE TITLE
Notice in php 8.1 autoloading causes psalm to fatal error

### ISSUE THREAD

**Issue by withinboredom** [vimeo/psalm#7691]
If any autoloaded files (in libraries) fire the notice:

```
Return type of Class:method() should either be compatible with Class::method: OtherClass, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in FILE:LINE in /app/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
```

then psalm fails to run any analysis at all. Obviously, libraries should update their code, but psalm should still be usable.

Psalm is installed via composer at version 4.20.0

**Comment by withinboredom** [comment]
> Hey @withinboredom, can you reproduce the issue on https://psalm.dev/ ?

No, because the file isn't autoloaded before analysis.

**Comment by withinboredom** [comment]
I ended up working around it by manually requiring the files and adding them to psalm as stubs.

**Comment by orklah** [comment]
When Psalm runs, it treats any notice as an error and crash. While it's okay for Psalm own code, when Psalm has to use a custom autoloader or runs code that has notices or warnings, it's sometimes an issue.

I guess we could declare an error handler each time we have to call the autoloader and then reverts back to the standard error handler.

Are you up for a PR?

**Comment by jmalinens** [comment]
I have similar issue with deprecations:

```
Uncaught Exception: RuntimeException PHP Error: User implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in User.php:27
Emitted in /var/lib/lighttpd/.composer/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:69
```

**Comment by orklah** [comment]
Yeah, that's expected given the explanation above. Feel free to push a fix or to ask for more details as to how it could be fixed!

**Comment by tolry** [comment]
One addition here: when I run psalm with `vendor/bin/psalm ... --threads=$(nproc)` within a github action, the fatal errors will be shown, the run continued and the command will return a `0` return code (indicating github actions that the run was successful) which is not what I would have expected.

If anyone can give me a hint on where the multi thread code lives, I can look into a small fix here.

**Comment by AndrolGenhald** [comment]
@tolry "threading" is actually done by forking, grep for `pcntl_fork`. I think src/Psalm/Internal/Fork/Pool.php is where you'll need to look.

**Comment by COil** [comment]
Same error here because of 'PHP deprecated' warnings in old PHP autoloaded librairies. I must disable `ErrorHandler::install($argv);` in `vendor/vimeo/psalm/src/Psalm/Internal/Cli/Psalm.php` in  to make it work. Is there a workaround for this? (can't use threads as a workaround)

**Comment by weirdan** [comment]
>  Is there a workaround for this? (can't use threads as a workaround)

None known. It's likely more code needs to use `ErrorHandler::runWithExceptionsSuppressed()`, but we'd have to see the backtrace to identify where it needs to be added.
